lesson26/src/js/login.js

```diff
@@ -38,18 +38,29 @@ const checkRegisteredUser = () => {
     }) 
 }
 
-const login = async () => {
+const awaitCheckRegisteredUser = async () => {
     try{
         const response = await checkRegisteredUser();
         if(!response) return;
-        localStorage.setItem("token",JSON.stringify(response.token));
-
-        window.location.href = "./index.html";
+        return response.token;
     }catch{
         window.location.href = "./401.html";
     }
 }
 
+const init = async () => {
+    const token = await awaitCheckRegisteredUser();
+    if(!token) return;
+    localStorage.setItem("token",JSON.stringify(token));
+}
```

Comment:
The work to check userData and the work to set the token were separated into two separate functions.